File: bot/src/main/java/com/ryderbelserion/discord/bot/configs/types/guilds/GuildCache.java
package com.ryderbelserion.discord.bot.configs.types.guilds;

import com.ryderbelserion.discord.api.utils.ConfigUtils;
import org.jspecify.annotations.NonNull;
import org.spongepowered.configurate.BasicConfigurationNode;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class GuildCache {

    private final List<String> guilds = new ArrayList<>();

    public void init(@NonNull final BasicConfigurationNode configuration) {
        final List<String> guilds = ConfigUtils.getStringList(configuration.node("whitelist"));

        this.guilds.clear();

        this.guilds.addAll(guilds);
    }

    public @NonNull final List<String> getGuilds() {
        return Collections.unmodifiableList(this.guilds);
    }
}

File: src/main/kotlin/com/ryderbelserion/bot/configs/types/GuildCache.kt
package com.ryderbelserion.bot.configs.types

import com.ryderbelserion.bot.util.ConfigUtil
import com.ryderbelserion.fusion.files.types.configurate.JsonCustomFile

class GuildCache(val file: JsonCustomFile) {

    private val guilds = arrayListOf<String>()

    fun init() {
        val configuration = file.configuration

        val list: List<String> = ConfigUtil.getStringList(configuration, "whitelist")

        this.guilds.clear()

        list.forEach {
            this.guilds.add(it)
        }
    }

    fun getGuilds(): ArrayList<String> {
        return this.guilds
    }
}